src/parser/datetime.rs

```diff
@@ -104,14 +194,112 @@ fn potential_interval_tokens(from: &DateTimeField) -> Vec<IntervalToken> {
+                (Colon, Colon) | (Plus, Plus) => { /* Matching punctuation */ }
+                (Dash, Dash) => {
+                    is_positive = false;
+                }
+                (Num(val), Num(_)) => {
+                    let val = *val;
+                    if hour_offset == -1 {
+                        // We haven't seen an hour offset yet so lets try to parse that
+                        if val <= 24 {
+                            hour_offset = val.try_into().unwrap();
+                        } else {
+                            return Err(ParserError::ParserError(format!(
+                                "Error parsing timezone string ({}): timezone hour invalid {}",
+                                value, val
+                            )));
+                        }
+                    } else if minute_offset == -1 {
+                        if val <= 60 {
+                            minute_offset = val.try_into().unwrap();
+                        } else {
+                            return Err(ParserError::ParserError(format!(
+                                "Error parsing timezone string ({}): timezone minute invalid {}",
+                                value, val
+                            )));
+                        }
+                    } else {
+                        // We've already seen an hour and a minute so we should never see another number
+                        return Err(ParserError::ParserError(format!(
+                            "Error parsing timezone string ({}): invalid value {} at token index {}", value,
+                            val, i
+                        )));
+                    }
+                }
+                (Zulu, Zulu) => return Ok(0 as i64),
+                (Name(val), Name(_)) => {
+                    // For now, we don't support named timezones
+                    return Err(ParserError::ParserError(format!(
+                            "Error parsing timezone string ({}): named timezones are not supported. Failed to parse {} at token index {}", value, val, i)));
+                }
+                (_, _) => {
+                    is_positive = true;
+                    hour_offset = -1;
+                    minute_offset = -1;
+                    break;
+                }
+            }
+        }
+
+        if hour_offset != -1 {
+            let mut tz_offset_second: i64 = hour_offset * 60 * 60;
+
+            if minute_offset != -1 {
+                tz_offset_second += minute_offset * 60;
+            }
+
+            if is_positive != true {
+                tz_offset_second *= -1
+            }
+            return Ok(tz_offset_second);
+        }
+    }
+
+    return Err(ParserError::ParserError(format!("It didnt work")));
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub(crate) enum IntervalToken {
     Dash,
     Space,
     Colon,
     Dot,
+    Plus,
+    Zulu,
     Num(u64),
     Nanos(u32),
+    Name(String),
```

Comment:
Could you either name this `TzName` or add a comment stating that it refers to a `TimeZone` (or a comment explaining what kinds of things it might refer to)